pyrobosim/pyrobosim/utils/pose.py

```diff
@@ -21,27 +31,36 @@ def __init__(self, x=0.0, y=0.0, z=0.0, roll=0.0, pitch=0.0, yaw=0.0, q=None):
         :type y: float
         :param z: Z position
         :type z: float
-        :param roll: Roll angle (about X axis), in radians
+        :param roll: Roll angle (about X axis), in specified angle units
         :type roll: float
-        :param pitch: Pitch angle (about Y axis), in radians
+        :param pitch: Pitch angle (about Y axis), in specified angle units
         :type pitch: float
-        :param yaw: Yaw angle (about Z axis), in radians
+        :param yaw: Yaw angle (about Z axis), in specified angle units
         :type yaw: float
         :param q: Quaternion, specified at [qw, qx, qy, qz]
             If specified, will override roll/pitch/yaw values.
         :type q: list[float]
+        :param angle_units: Units for angle ('radians' or `degrees`). Default is 'radians'
+        :type angle_units: str
         """
+        if angle_units not in ("radians", "degrees"):
+            raise ValueError("Angles should be either in 'radians' or 'degrees'.")
+
         self.x = x
         self.y = y
         self.z = z
+        self.angle_units = angle_units
+
+        if self.angle_units == "degrees":
```

Comment:
I agree with you. If user specify quaternion input, it does not make sense to execute code related to angles. 

Hence I am thinking this logic:
- return early if `q` provided.
- otherwise check `angle_units` and do needful as per input value (either `radians` or `degrees`)

```
    if q is not None:
        self.set_quaternion(q)
        return

    if angle_units not in ("radians", "degrees"):
        raise ValueError(
            "Invalid angle units provided. It should be either 'radians' or 'degrees'."
            "Additionally, if 'q' is provided, angle units are ignored."
        )

    self.angle_units = angle_units

    if self.angle_units == "degrees":
        roll, pitch, yaw = map(math.radians, [roll, pitch, yaw])

    self.set_euler_angles(roll, pitch, yaw)

```